shell/browser/api/atom_api_web_request.cc

```diff
@@ -84,7 +68,24 @@ void WebRequest::SetListener(Method method, Event type, mate::Arguments* args) {
   // { urls }.
   URLPatterns patterns;
   mate::Dictionary dict;
-  args->GetNext(&dict) && dict.Get("urls", &patterns);
+  if (args->GetNext(&dict)) {
+    std::set<std::string> filter_patterns;
+    if (dict.HasKey("urls") && dict.Get("urls", &filter_patterns)) {
+      for (const std::string& filter_pattern : filter_patterns) {
```

Comment:
_(minor)_ This block nests the control flow four layers deep -- if-if-for-if -- making the code slightly more complicated without any benefit. If the "get the patterns" and "parse the patterns" steps are separated, the code becomes simpler and loses two nesting layers:

```c++
// get the filter patterns
std::set<std::string> filter_patterns;
mate::Dictionary dict;
args->GetNext(&dict) && dict.Get("urls", &filter_patterns);

// parse the filter patterns
URLPattern pattern (URLPattern::SCHEME_ALL);
for (const std::string& filter_pattern : filter_patterns) {
  const URLPattern::ParseResult result = pattern.Parse(...);
  if (result == URLPattern::ParseResult::kSuccess) ...
}
```